c/restService.c

```diff
@@ -0,0 +1,266 @@
+
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+*/
+
+
+#ifdef METTLE
+#include <metal/metal.h>
+#include <metal/stddef.h>
+#include <metal/stdio.h>
+#include <metal/stdlib.h>
+#include <metal/string.h>
+#include "metalio.h"
+#include "qsam.h"
+#else
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <strings.h>
+#endif
+#include <sys/utsname.h>
+
+#include "zowetypes.h"
+#include "bpxnet.h"
+#include "utils.h"
+#include "socketmgmt.h"
+
+#include "httpserver.h"
+#include "json.h"
+#include "logging.h"
+#include "zssLogging.h"
+#include "restService.h"
+#pragma linkage (EXSMFI,OS)
+
+#ifdef __ZOWE_OS_ZOS
+static int serveStatus(HttpService *service, HttpResponse *response);
+static JsonObject* serverConfig = NULL;
+char productVersion[40];
+extern char **environ;
+
+typedef int EXSMFI();
+EXSMFI *smf_func;
+
+int installServerStatusService(HttpServer *server, JsonObject *serverSettings, char* productVer)
+{
+  HttpService *httpService = makeGeneratedService("REST_Service", "/server/agent/**");
+  httpService->authType = SERVICE_AUTH_NATIVE_WITH_SESSION_TOKEN;
+  httpService->serviceFunction = &serveStatus;
+  httpService->runInSubtask = TRUE;
+  httpService->doImpersonation = TRUE;
+  registerHttpService(server, httpService);
+  serverConfig = serverSettings;
+  memcpy(productVersion, productVer, 40);
```

Comment:
I should verify that it is null terminated but this line is how I know it will be equal to 40 bytes: https://github.com/zowe/zss/blob/master/c/zss.c#L84

Reply:
This can be best solved by using a `#DEFINE` to state that both are 40.